fnext/listeners.go

```diff
@@ -17,9 +17,9 @@ type AppListener interface {
 	// AfterAppUpdate called after updating App in the database
 	AfterAppUpdate(ctx context.Context, app *models.App) error
 	// BeforeAppDelete called right before deleting App in the database
-	BeforeAppDelete(ctx context.Context, app *models.App) error
+	BeforeAppDelete(ctx context.Context, appName string) error
 	// AfterAppDelete called after deleting App in the database
-	AfterAppDelete(ctx context.Context, app *models.App) error
+	AfterAppDelete(ctx context.Context, appName string) error
```

Comment:
I think it would be useful for to have all of the application info here. Why the change?

Reply:
well, it means we have to go fill it in instead of just deleting it (which we weren't / aren't, which is misleading at least), the db call itself to delete doesn't fetch and this turns into a plumbing thing. I can't think of other fields in the app itself that would cause any practical use cases to prevent deletion really... 'don't delete apps if they have a config value with key=AZEROTH' ?

Reply:
> small change to BeforeAppDelete/AfterAppDelete -- we were passing in a half
baked struct with only the name filled in and not filling in the fields
anywhere. this is mostly just misleading, we could fill in the app, but we
weren't and don't really want to, it's more to notify of an app deletion event
so that an extension can behave accordingly instead of letting a user inspect
the app. i know of 3 extensions and the changes required to update are very
small.

(from PR comment)